src/blocks/networkmanager.rs

```diff
@@ -784,7 +826,53 @@ impl Block for NetworkManager {
             }
         };
 
-        Ok(None)
+        /* TO-DO: Rustify */
+        let next_update = match self.external_ip_format.has_tokens() {
+            true => {
+                let external_ip = {
+                    let ip_info = match http::http_get_json(
+                        API_ENDPOINT,
+                        Some(Duration::from_secs(3)),
+                        vec![],
+                    ) {
+                        Ok(ip_info_json) => serde_json::from_value(ip_info_json.content).unwrap(),
+                        _ => IPAddressInfo::default(),
+                    };
+                    let flag = country_flag_from_iso_code(ip_info.country_code.as_str());
+                    let values = map!(
+                        "address" => Value::from_string (ip_info.address),
+                        "country" => Value::from_string (ip_info.country),
+                        "country_code" => Value::from_string (ip_info.country_code),
+                        "region" => Value::from_string (ip_info.region),
+                        "region_name" => Value::from_string (ip_info.region_name),
+                        "city" => Value::from_string (ip_info.city),
+                        "zip" => Value::from_string (ip_info.zip),
+                        "latitude" => Value::from_float (ip_info.lat),
+                        "longitude" => Value::from_float (ip_info.lon),
+                        "timezone" => Value::from_string (ip_info.timezone),
+                        "isp" => Value::from_string (ip_info.isp),
+                        "org" => Value::from_string (ip_info.org),
+                        "autonomous_system" => Value::from_string (ip_info.autonomous_system),
+                        "country_flag" => Value::from_string(flag),
+                    );
+                    if let Ok(s) = self.external_ip_format.render(&values) {
+                        s.0
+                    } else {
+                        "[invalid external_ip format string]".to_string()
```

Comment:
It might be better to propagate the error (and thus telling the user what's the problem).

Reply:
How? That check is inside a block that returns an Option, not an Error. I guess that's what the other blocks do the same thing.